### DISCUSSION TITLE
Shall this work eventually be upstreamed?

### DISCUSSION THREAD

**Discussion by roke-julian-lockhart** [uutils/coreutils#3712] [Q&A] [answered]
I wonder whether you, developers, intend to upstream the work at all. Not merely because this project is being modernly rewritten in Rust, but, primarily, because *it is cross-platform*!

I realize that convincing the GNU Project might be the most difficult problem to overcome, but I worry that this project shall remain niche if it eternally remains a competitor to its predecessor.

(As you probably estimated, I wanted to enable Discussions to ask this question. Thanks for that.)

**Comment by tertsdiepraam** [comment, marked as answer]
I really doubt that the GNU project will want to upstream it. The most important reason is that they probably don't like the license (uutils is MIT, GNU always uses GPL and we can't change it without asking every contributor who has a line of code in the latest version).

From our side, the GNU project adheres to a philosophy of software that our current contributors might not agree with. So, I think coexistence is the goal in general.

But, that doesn't mean that it might not get adopted by Linux distros at some point as the default set of coreutils. Or that it becomes common to use uutils on Windows. We'll have to see what the future holds :)

**Reply by roke-julian-lockhart** [reply]
I'd absolutely love for it to become the default set of coreutils. I expect that the maintainers of [`microsoft/CBL-Mariner`](https://github.com/microsoft/CBL-Mariner) would adore this project if it were complete.

**Reply by FruityWelsh** [reply]
Wouldn't you be able to dual license with everything historical being MIT, but reused in a GPL project, with every thing after the adoption of the new license being GPL?

**Reply by tertsdiepraam** [reply]
Maybe, but we'd rather just continue as it is.

**Reply by FruityWelsh** [reply]
Unfortunate, but I respect your work regardless!

**Reply by roke-julian-lockhart** [reply]
> I really doubt that the GNU project will want to upstream it. The most important reason is that they probably don't like the license (uutils is MIT, GNU always uses GPL and we can't change it without asking every contributor who has a line of code in the latest version).

> But, that doesn't mean that it might not get adopted by Linux distros at some point as the default set of coreutils. Or that it becomes common to use uutils on Windows.

@tertsdiepraam, do you have any *intention* of trying to get your rewrite into OSes more accepting of FOSS-licensed code that doesn't limit what consumers can relicense it as?

As an example, Fedora (arguably the most popular distribution for *desktop* users) explicitly permits every variant of MIT in existence. [^1]

[^1]: [`docs.fedoraproject.org/en-US/legal/allowed-licenses`](https://docs.fedoraproject.org/en-US/legal/allowed-licenses/#:~:text=Glyph%20List%20License-,MIT,-ADSL)

> https://github.com/uutils/coreutils/discussions/3712#discussioncomment-5017308

Continuing with MIT is probably the best idea. GPL basically makes the project untouchable for most organisations.
